### DISCUSSION TITLE
Jan: Customizing and saving prompts

### DISCUSSION THREAD

**Discussion by eckartal** [janhq/jan#3624] [Feature Ideas]
## Overview
People use specific models for specific reasons and need to send the initial prompt first to set the tone again and again. We should have a feature that allows users to save prompts for specific models. 

## Goals

- Implement a system to save and manage custom prompts
- Enable easy application of saved prompts to specific models
- Allow users to categorize and organize their saved prompts

**Comment by rvrstfn** [comment]
This should be coupled with a simple system that allows to insert "variables" into the prompt.

Nothing fancy, a simple {{variable}} would do.

The "variables" would be the additional services provided by Jan: RAG results, web search results, persistent memories, user biography, tone settings... Or anything user-defined.

**Reply by dan-menlo** [reply]
This is a great idea and would be incredibly useful!
